sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -49,38 +49,30 @@ protected StringComparison CaseSensitivity
 
         protected abstract bool IsStatic(TMemberDeclarationSyntax member);
 
+        protected abstract bool IsAbstract(TMemberDeclarationSyntax member);
+
         protected void CheckMembers(SyntaxNodeAnalysisContext c, IEnumerable<TMemberDeclarationSyntax> members)
         {
-            var misplacedOverloadsMapping = GetMisplacedOverloads(members);
-            foreach (var misplacedOverloadsEntry in misplacedOverloadsMapping)
+            var misplacedOverloads = GetMisplacedOverloads(c, members);
+            foreach (var misplacedMethods in misplacedOverloads.Values.Where(x => x.Count > 1))
             {
-                var misplacedOverloadsByAccessibility = misplacedOverloadsEntry.Value
-                    .GroupBy(memberDeclaration => c.SemanticModel.GetDeclaredSymbol(memberDeclaration)?.DeclaredAccessibility);
-
-                foreach (var misplacedOverloads in misplacedOverloadsByAccessibility)
-                {
-                    var firstMethodSignature = misplacedOverloads.First();
-                    if (misplacedOverloads.Key != null && misplacedOverloads.Count() > 1 && firstMethodSignature != null)
-                    {
-                        var secondaryLocations = misplacedOverloads
-                            .Skip(1)
-                            .Select(member => GetNameSyntaxNode(member))
-                            .Where(nameSyntax => nameSyntax != null)
-                            .Select(nameSyntax => new SecondaryLocation(nameSyntax?.GetLocation(), "Non-adjacent overload"));
-                        c.ReportDiagnosticWhenActive(
-                            Diagnostic.Create(
-                                descriptor: Rule,
-                                location: GetNameSyntaxNode(firstMethodSignature)?.GetLocation(),
-                                additionalLocations: secondaryLocations.ToAdditionalLocations(),
-                                properties: secondaryLocations.ToProperties(),
-                                messageArgs: GetMethodName(firstMethodSignature, true)));
-
-                    }
-                }
+                var firstMethod = misplacedMethods.First();
+                var secondaryLocations = misplacedMethods
+                    .Skip(1)
+                    .Select(member => GetNameSyntaxNode(member))
+                    .Where(nameSyntax => nameSyntax != null)
```

Comment:
we have an extension method `WhereNotNull`

Reply:
The existing one did not work for struct?, I've added an overload.